hudi-common/src/main/java/org/apache/hudi/common/table/view/FileSystemViewManager.java

```diff
@@ -220,16 +220,15 @@ public static FileSystemViewManager createViewManager(final HoodieEngineContext
                                                         final HoodieMetadataConfig metadataConfig,
                                                         final FileSystemViewStorageConfig config,
                                                         final HoodieCommonConfig commonConfig) {
-    return createViewManager(context, metadataConfig, config, commonConfig, (SerializableSupplier<HoodieTableMetadata>) null);
+    return createViewManager(context, metadataConfig, config, commonConfig, null);
   }
 
-  public static FileSystemViewManager createViewManager(final HoodieEngineContext context,
-                                                        final HoodieMetadataConfig metadataConfig,
-                                                        final FileSystemViewStorageConfig config,
-                                                        final HoodieCommonConfig commonConfig,
-                                                        final String basePath) {
+  public static FileSystemViewManager createViewManagerWithTableMetadata(final HoodieEngineContext context,
+                                                                         final HoodieMetadataConfig metadataConfig,
+                                                                         final FileSystemViewStorageConfig config,
+                                                                         final HoodieCommonConfig commonConfig) {
     return createViewManager(context, metadataConfig, config, commonConfig,
```

Comment:
Not sure if we need to differentiate the `metadataConfig` across tables as well.  For now, it's fine.

Reply:
I think we would need a way to get this config at runtime but I don't see a way to do that with the current structure

Reply:
Typically metadata config is controlled internally. When there are metadata configs that could differ by table, then I think it should be part of table config (e.g. the metadata partitions that are enabled is already part of table config). Since we have the metaClient here, we can get the tableConfig as well.